pytorch_lightning/trainer/trainer.py

```diff
@@ -781,6 +782,135 @@ def slurm_job_id(self) -> int:
             job_id = None
         return job_id
 
+    @staticmethod
+    def add_default_args(parent_parser):
+
+        parser = ArgumentParser(parents=[parent_parser])
```

Comment:
these should be auto-generated given a trainer class. we don't want to start making sure this list maintains parity.

@Borda

Reply:
Also, could we add it to another mixin? This file is already very long.

Reply:
we can regenerate it from class attributes, we may also get types (not sure hot the typing would cooperate)
but not sure how to pass help strings and single-letter shortcuts in an argument... 

@PyTorchLightning/core-contributors any idea about passing help string
otherwise I would go for the automatic generated, good idea

Reply:
I've drafted a solution to this that automatically parses __intit__'s doc string for the help fields and the functions signature to populate the argument parser

Should I make a seperate pull request?